pkg/rbd/rbd_util.go

```diff
@@ -813,3 +840,81 @@ func cleanupRBDImageMetadataStash(path string) error {
 
 	return nil
 }
+
+func flattenRbdImage(ctx context.Context, rbdVol *rbdVolume, hardmaxDepth, softmaxDepth uint, cr *util.Credentials) error {
+
+	depth, err := getCloneDepth(ctx, rbdVol.Monitors, rbdVol.Pool, rbdVol.RbdImageName, 0, cr)
+	if err != nil {
+		return err
+	}
+
+	klog.Infof(util.Log(ctx, "image depth is %v and maximum configured clone depth hard limit is %v and soft limit is %v"), depth, hardmaxDepth, softmaxDepth)
+	if depth >= int(hardmaxDepth) {
+		klog.Infof(util.Log(ctx, "maximum clone depth (%d) has been reached, flattening %v volume "), hardmaxDepth, rbdVol.RbdImageName)
+		args := []string{"-m", rbdVol.Monitors,
```

Comment:
CSI spec does not mention anything related to  in-progress response

Reply:
I thought CephFS was (or was planning) to send some type of "in-progress" when it was cloning (i.e. slowly copying)?

Reply:
I don't think that is possible. need a second pair of Eyes  can you please confirm  it once https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume

Reply:
Looking at the external-provisioner sidecar [1], it looks like you can return numerous errors to tell it it's pending [2]


[1] https://github.com/openshift/csi-external-provisioner/blob/master/pkg/controller/controller.go#L542
[2] https://github.com/openshift/csi-external-provisioner/blob/master/pkg/controller/controller.go#L1139

Reply:
... specifically the `ABORTED` error code looks like it should be used

Reply:
yeah, this can be used, working late night forgot about ABORT at all :D

once we reach the hard limit we need to check the flatten status right (if it's not done need to  return Abort error message)? can you provide me the command to check task status?

Reply:
I think I can use `ceph rbd task list`  if the rbd image name is  present on this list means the task is not completed and have to return  `ABORT` error message
if the task is completed  means return  success
@dillaman let me know if my assumption is correct will implement this.

Reply:
Once flatten is complete, you wouldn't even enter this check, right? Since you would have gotten the clone depth and it should return 0. Otherwise, feel free to keep running "ceph rbd task add flatten" since it will just keep returning the original task id back to you if it already existed (i.e. no need to parse the list I don't think).